### PR TITLE
Task 2

### LINKED ISSUE #558: gridlayout::AMRToLocal signedness
should probably only accept signed ints. and should probably only return unsigned ints

### PR DESCRIPTION
**Pull request by alexandre-de-larminat**
Fixes #558 using Philip's edits. Unsure how these require only signed ints though - shouldn't something like `static_assert(std::is_integral_v<T> && std::is_signed_v<T>, "Error, must be MeshIndex (integral Point)");` be required?

## Summary by CodeRabbit

- **Refactor**
  - Updated the grid layout handling to use a standard unsigned integer type for improved reliability.

- **Chores**
  - Updated a subproject to the latest commit for enhanced functionality and fixes.

- **Tests**
  - Enhanced a test script with a timestamp in the figure title for better traceability of test results.

### CODE REVIEW

File: src/core/data/grid/gridlayout.hpp [hunk 1]
```diff
@@ -796,15 +796,17 @@ namespace core
...
             //
             for (auto i = 0u; i < dimension; ++i)
             {
-                localPoint[i] = AMRPoint[i] - (AMRBox_.lower[i] - localStart);
+                int local = AMRPoint[i] - (AMRBox_.lower[i] - localStart);
+                assert(local >= 0);
+                localPoint[i] = local;
             }
             return localPoint;
         }
```
[the remark this answers is not in the document]
Comment (PhilipDeegan) [reply src/core/data/grid/gridlayout.hpp:812]: @alexandre-de-larminat don't mind this, it's not always accurate.

**Comment by PhilipDeegan** [comment]
you will need to make this sort of change too https://github.com/PhilipDeegan/PHARE/blob/f59ede6870a7203d0514f3530518955c67454a1d/tests/core/data/gridlayout/gridlayout_amr.cpp#L87

**Comment by alexandre-de-larminat** [comment]
Hi, thanks for your suggestions. Could you explicit what you mean when you suggest I rebase and drop the commits—should I revert the changes I made and then rebase?

**Comment by PhilipDeegan** [comment]
@alexandre-de-larminat 

you should try something like this

```
# checkout new branch
git checkout -b uint_amr_to_local
# enter interactive mode to drop some commits
git rebase -i HEAD~11
```
then you should make it look like this 

image

then sync with main repo, and push to your remote

```
# sync with main repo
git pull https://github.com/PHAREHUB/PHARE master --rebase 
# push to your remote, assuming origin == https://github.com/alexandre-de-larminat/PHARE
git push origin uint_amr_to_local                                                    
```

then make a new pull request on the new branch

and it should look something like this, with only one commit

https://github.com/PhilipDeegan/PHARE/pull/76

**Comment by PhilipDeegan** [comment]
@alexandre-de-larminat I think it's more important that it returns a `uint`, and not so important that it must accept a `int` so don't worry about that part for the moment, I'm not sure it matters what the input type is now that I think about it

**Comment by PhilipDeegan** [comment]
:rabbit: giving a different response to the same [code](https://github.com/PhilipDeegan/PHARE/pull/76#discussion_r1393882948)...

**Pull request closed without merging**